packages/botonic-nlu/src/botonic-nlu.ts

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { shuffle } from './util/object-tools'
+import { Preprocessor } from './preprocessor'
+import { ModelManager } from './model-manager'
+import { IntentsProcessor } from './intents-processor'
+import { DataReader } from './data-reader'
+import {
+  DecodedPrediction,
+  WordEmbeddingType,
+  WordEmbeddingDimension,
+  DataSet,
+  InputSet,
+  OutputSet,
+  ModelTemplatesType,
+  Vocabulary,
+  PreprocessorEngines,
+  Tokenizer,
+  Normalizer,
+  Stemmer,
+} from './types'
+import { Language } from './language'
+import { readJSON } from './util/file-system'
+import { LayersModel, Sequential, tensor } from '@tensorflow/tfjs-node'
+import { join } from 'path'
+import { MODELS_DIR, MODEL_DATA_FILENAME, NLU_DIR } from './constants'
+import { mkdirSync, writeFileSync } from 'fs'
+import { DefaultTokenizer } from './preprocessing-tools/tokenizer'
+import { DefaultNormalizer } from './preprocessing-tools/normalizer'
+import { DefaultStemmer } from './preprocessing-tools/stemmer'
+
+export class BotonicNLU {
+  language: Language
+  maxSeqLen: number
+  vocabulary: Vocabulary
+  engines: PreprocessorEngines = {}
+  private modelManager: ModelManager
+  private preprocessor: Preprocessor
+  private intentsProcessor: IntentsProcessor
+  private dataReader = new DataReader()
+
+  constructor({
+    normalizer = new DefaultNormalizer(),
+    tokenizer = new DefaultTokenizer(),
+    stemmer = new DefaultStemmer(),
+  }: {
+    normalizer?: Normalizer
+    tokenizer?: Tokenizer
+    stemmer?: Stemmer
+  }) {
```

Comment:
You could add a default empty object to allow instantiating `BotonicNLU()` instead of `BotonicNLU({})`:

```
constructor({
    normalizer = new DefaultNormalizer(),
    tokenizer = new DefaultTokenizer(),
    stemmer = new DefaultStemmer(),
  }: {
    normalizer?: Normalizer
    tokenizer?: Tokenizer
    stemmer?: Stemmer
  } = {}) {
    ...
  }
```